src/Sulu/Bundle/DocumentManagerBundle/DataFixtures/DocumentFixtureLoader.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Sulu\Bundle\DocumentManagerBundle\DataFixtures;
+
+use Sulu\Bundle\DocumentManagerBundle\DataFixtures\DocumentFixtureInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * This class locates and sorts fixture files
+ */
+class DocumentFixtureLoader
+{
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * Load, instantiate and sort all fixture files found
+     * within the given paths.
+     *
+     * @param array $paths
+     *
+     * @return DocumentFixtureInterface[]
+     */
+    public function load(array $paths)
+    {
+        $finder = new Finder();
+        $finder->in($paths);
+        $finder->name('*Fixture.php');
+
+        foreach ($finder as $file) {
+            $declaredClasses = get_declared_classes();
+            require_once($file);
+            $declaredClassesDiff = array_diff(get_declared_classes(), $declaredClasses);
+            $fixtureClass = array_pop($declaredClassesDiff);
```

Comment:
one class you mean? yeah. tbh determining the class from a file is non-trivial. The above method will not work twice in the same request for instance. The doctrine fixtures bundle just traverses ALL of the loaded classes and instanatiates ALL which implement the DoctrineFixturesInterface. But that approach caused issues here with prophecy.

I think this method will be ok.